### PR TITLE
Fonts loading strategy refactor

### PR DESCRIPTION
**Pull request by svillafe**
## Why are you doing this?

In order to improve the performance of our site, we are updating the font-loading strategy. The new strategy is based on [this blog post](https://goo.gl/n9qKMg)

[**Trello Card**](https://trello.com/c/Ak95VIT5/330-fonts-refactor-it-could-give-us-as-much-as-12-seconds)

## Changes

* Removed the fonts from CSS.
* Added `woff2`, `woff` and `ttf` fonts. So the browser will download the best font for it.
* Added `preload` to push to the front of the queue the downloading process of the fonts (this is available in some browsers)

## Impact

picture 751

picture 752

picture 753

## Screenshots

|                       | Before | After |
|-----------------------|--------|-------|
| `/uk`   | fonts-before-uk  | fonts-after-uk|
| `/uk/contribute`      |    fonts-before-uk-contribute | fonts-after-uk-contribute|
| `/contribute/one-off` |   fonts-before-uk-one-off | font-after-uk-oneoff|

### CODE REVIEW

File: assets/fonts/fonts.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require('./GHGuardianHeadline-Bold.woff');
```
Comment (sndrs) [inline assets/fonts/fonts.js]: does this mean everyone wil have to download all fonts?

Comment (sndrs) [reply assets/fonts/fonts.js]: actaully - what does this do? i'm not sure?

Comment (svillafe) [reply assets/fonts/fonts.js]: In order to tell webpack to include the fonts in the `compile-assets`, we provide this file as an entry point.

Comment (svillafe) [reply assets/fonts/fonts.js]: We do that in [this line](https://github.com/guardian/support-frontend/pull/634/files#diff-5c572bb8f24a00a84aa28f22d3a39ba5R24).

**Comment by Ap0c** [comment]
Do we need `ttf`? I think [every browser](https://caniuse.com/#feat=woff) we support can handle `woff` at minimum.

**Comment by svillafe** [comment]
@Ap0c I set it up as a default, the browser will only download `ttf` only when `woff2` or `woff` is not supported by it.

File: assets/stylesheets/garnett.scss [hunk 2]
```diff
@@ -7,38 +7,50 @@
...
...ORrQ/XUdT6HvWlM1ha9gedobdYPdZEsvnZTnlA/koPplv4ydEEUFFLdFB9BGTxTJxVFwVyeKVTvXpepKeb5Q2iBEazYx+xkLjkdnQjMx+5nDznBlrfrWAZVrHrdtWvPXM+mqXsWvbkT3Onmuvst86dZ3Bzlan0PnslnZrug3dyO3vznDXuVfcNK+0x7zQ2+4X83v48/39fuH/X4JKQf2gazAoOPALEkrNEAAAeAGcmQVgW0fS+HffE9gCW9KzJMuSJUuywCTZliWZWWaOY8mcxHG43CRlpiTFpMyQfr1y6yZ3/zIc9LvS8fXK7fGlzHR677+772klJ+5HBcPPTzOzM7O7M/MAA7IBYO5jfgNyAQeAx+dUmo3ucJQLcWajM8yyIRZ+38ALjSMvtP5r5FAr//3mo97aVrW0pC9y6phbk2t0ziKmlX8UdgNBEGWxv2C8AAIAlHAC9IH7V+CT4KwMvkS5GoyDA5RfTfk8iIM7KT+H8jhYA/ZR/lvKZ8F6cA/li5QnwGpwDeXXUj6N9D6AfSGEiS+KQDEAcpc3XBOJLvOIW6Ew5plC1ZGoUaEgzvlHRW1Tb68AUg7aWVtuLdAaNdrSp5GjXvR7HE3+tzK85Q+ZjUZWVpylYsuw21LWyB6UrPGJ1hMvnQuvBiywoL+vQVZxwA58KEZ5CrczHNX7ws5qkzFPoTSaTB5kUrgmwLjdevKT1+1m1px/Mf/CZBwWXHDWwPimWf60ePuU0d/gmThqND4xdNxTM+ucemabwTk+0zitVWqnTmsarojaVck9LSPjQwDZYhe+YPqZt4AT6UQrjoaUCoXbFWDCNc1MqNpkNrt9Xq/bhT0CTyqPmQqDMt9i78CZiaqGtWd0ti04ZcEifcxzwvatRbnqKw7qDKWJi9du3pvw5aoeujBbe+ctt98rCEiLHmtBq9ei1Svg1CwgFOkmEdVKGXMVeJTypyhXg03gGcovp3wGbAdXUf4s5RqwDRxEkTYjn96OfKrB3oXIj9SpsLqZkVwJq55eOrDEvzGZgN6GRH2hq3WuYYz5zY9vPfA8ct1Remdyuqjv1Kl1p3Zakhdiq4lU9iDSxkna1oK7ASQRjCFt6O8hMXLId059KBoyun1uvRO51pKIM7lZgSL+u8sLfL6CyyHnC6iQlq2c88Fz+H8sfLXI//qWlwRBlEX2klHaY/Xg/hX4JDg+gy9
```
Comment (Ap0c) [inline assets/stylesheets/garnett.scss]: This code can probably go into `webfonts.scss` now and we can finally delete `garnett.scss` 🎉!

File: webpack.common.js [hunk 3]
```diff
@@ -64,6 +65,10 @@ module.exports = (cssFilename, outputFilename, minimizeCss) => ({
         test: /\.(png|jpg|gif|ico)$/,
         loader: 'file-loader?name=[path][name].[hash].[ext]',
       },
+      {
+        test: /\.(eot|ttf|woff|woff2)$/,
```
Comment (Ap0c) [inline webpack.common.js]: We don't have any `eot` fonts do we?

File: assets/stylesheets/garnett.scss (hunk 2 above)
Comment (svillafe) [reply assets/stylesheets/garnett.scss]: That is great, I will add it to the CSS refactor PR that I am preparing.

**Comment by paperboyo** [comment]
> Just wondering if this means we'll be able to use fonts loaded and cached on theguardian.com

That would be ideal, but I don’t think this has PR has anything to do with it. Currently, this may not be possible as dotcom keeps them in local storage and it may not be possible to share it between domains. @sndrs and/or @regiskuckaertz would defo know more, though.

**Comment by GHaberis** [comment]
@mateusz is correct, we're not downloading font files on theguardian.com, we're retrieving encoded fonts and storing them in localStorage on first visit and retrieving them on subsequent visits.

**Comment by regiskuckaertz** [comment]
I think this is awesome given how much the brand relies on typography 💥 The `localStorage` solution is extremely clever and still top of the class, but one would hope the browser could provide a declarative equivalent and deal with implementation details itself. The Web is [still 95% typography](https://ia.net/topics/the-web-is-all-about-typography-period) after all.

**Comment by svillafe** [comment]
It is working in `PROD`, the post-deploy tests are broken.

**Comment by sndrs** [comment]
> It is working on PROD, the post-deploy tests are broken.

👌

**Pull request merged**